webcurator-webapp/src/main/java/org/webcurator/rest/Targets.java

```diff
@@ -668,7 +668,7 @@ public HashMap<String, Object> errorMessage(MethodArgumentNotValidException ex)
     private SearchResult search(Filter filter, Integer offset, Integer limit, String sortBy) throws BadRequestError {
 
         // defaults
-        if (limit == null) {
+        if (limit == null || limit <= 0) {
```

Comment:
I agree. I would even go slightly further and let only "limit == -1" return everything and let "limit >= 0" retain its original semantics, because I think being able to download zero records can be useful in some scenarios.

Reply:
@jeremy-liss @hannakoppelaar Thank you for the feedback. I've update the PR to fix the meaning of 'limit':

- If limit is null, set it to the DEFAULT_PAGE_LIMIT ( DEFAULT_PAGE_LIMIT=10).
- If limit is -1, set it to Integer.MAX_VALUE.
- Then if limit <=0, thows an exception.

@hannakoppelaar Because 'limit' is used as the divisor to calculate how many pages. So it's value can't be 0. 

I'll checkout and update these changes to https://github.com/WebCuratorTool/webcurator/pull/149 as well. The changes of the server api on #149 will be kept when merging.

Reply:
You're right, I hadn't realized limit was being used as a divisor!